components/utils/useSessionGroups.ts

```diff
@@ -0,0 +1,181 @@
+import { Moment } from 'moment'
+import React from 'react'
+import { Session } from '../../config/types'
+import dateTimeProvider from './dateTimeProvider'
+
+type SessionId = string // UUID
+
+export interface SessionGroup {
+  timeStart: Moment
+  timeEnd: Moment
+  sessions: Session[]
+}
+
+interface SessionGroups {
+  allSessionGroups: SessionGroup[]
+  currentSessionGroup?: SessionGroup
+  previousSessionGroup?: SessionGroup
+  nextSessionGroup?: SessionGroup
+  pastSessionGroups?: SessionGroup[]
+  futureSessionGroups?: SessionGroup[]
+}
+
+function getSessionById(sessions: Session[], ids: SessionId[]) {
+  return sessions.filter(session => ids.includes(session.Id))
+}
+
+// Creates the session groups.
+// Important: Order is assumed to be correct.
+function createSessionGroups(conferenceDate: Moment, sessions: Session[]): SessionGroup[] {
```

Comment:
This stuff is specific to this year's conference so shouldn't appear in `components`. Suggestion would be to add it to the `Conference` object?

We'll then need to figure out how to incorporate that into our previous agenda stuff later in the year, but that should be fine.

Reply:
Good point, not the best home for this. Ideally it would set the agenda page up too but I'm not sure how to structure that yet with breaks etc. Perhaps more thought is needed on the data structure